plutus-pab/src/Plutus/PAB/Webserver/Server.hs

```diff
@@ -158,10 +158,15 @@ startServer' waiMiddleware port walletClient staticPath availability = do
     logInfo @(LM.PABMultiAgentMsg t) (LM.StartingPABBackendServer port)
     void $ liftIO $
         forkFinally
-            (Warp.runSettings warpSettings $ waiMiddleware $ app staticPath walletClient simRunner)
+            (Warp.runSettings warpSettings $ applyMiddlewares waiMiddlewares
+               $ app staticPath walletClient simRunner)
             (\_ -> putMVar mvar ())
 
     pure (mvar, liftIO $ STM.atomically $ STM.putTMVar shutdownVar ())
+ where
+  applyMiddlewares :: [Middleware] -> Application -> Application
```

Comment:
`appEndo . foldMap Endo` would also do for fusing the middlewares together (using the `Monoid` instance of `Endo`)

Reply:
Do you think that is more readable?

Reply:
Yes, and it applies them in the correct order too - which does make a difference since `Middleware` is not commutative. I'll put up a fix.